src/modules/storage/LedgerStorage.ts

```diff
@@ -1019,20 +1019,48 @@ export class LedgerStorage extends Storages
      */
     public getUTXO (address: string): Promise<any[]>
     {
-        let sql =
+        // TODO We should apply the block's timestamp to this method when it is added
+        let sql_utxo =
             `SELECT
-                tx_outputs.utxo_key AS utxo,
-                transactions.type,
-                transactions.unlock_height,
-                tx_outputs.amount
+                O.utxo_key as utxo,
+                O.amount,
+                T.block_height,
+                STRFTIME('%s', '2020-01-01 00:00:00') * 1000 + T.block_height * 10 * 60000 as block_time,
+                T.type,
+                T.unlock_height
             FROM
-                tx_outputs,
-                transactions
+                tx_outputs O
+                INNER JOIN transactions T ON (T.tx_hash = O.tx_hash)
+                INNER JOIN blocks B ON (B.height = T.block_height)
             WHERE
-                tx_outputs.tx_hash = transactions.tx_hash
-                AND tx_outputs.address = ?
-                AND tx_outputs.used != 1;`;
-        return this.query(sql, [address]);
+                O.address = ?
+                AND O.used = 0;
+            ORDER BY T.block_height, O.amount
+            `;
+
+        let sql_pending =
+            `SELECT
+                S.utxo_key as utxo
+            FROM
+                tx_outputs S
+                INNER JOIN tx_input_pool I ON (I.utxo = S.utxo_key)
+                INNER JOIN transaction_pool T ON (T.tx_hash = I.tx_hash)
+            WHERE
+                S.address = ?;
+           `;
+
+        let result: any[];
+        return new Promise<any[]>((resolve, reject) => {
+            this.query(sql_utxo, [address])
+                .then((utxos: any[]) => {
+                    result = utxos;
+                    return this.query(sql_pending, [address]);
+                })
+                .then((pending: any[]) => {
+                    resolve(result.filter(n => pending.find(m => (Buffer.compare(n.utxo, m.utxo) === 0)) === undefined));
```

Comment:
What happens if `utxos` is null?

Reply:
If the value is null, it means an error has occurred and it moves to catch.

Reply:
So I don't think we need any more code.